### PR TITLE
Update Font Awesome to version 5.12.1

### LINKED ISSUE #1278: Update Font Awesome to version 5.12.1
Storefront is still using Font Awesome 5.8.1, although Font Awesome 5.12.1 is available since February 4, 2020.

In https://github.com/woocommerce/storefront/issues/1089, improving the Font Awesome integration had been requested by several people. However, as Storefront uses the SCSS files instead of the CSS file, including the latest Font Awesome CSS file remote, as suggested in that issue, will not work. Therefore, at least the library should be updated to the latest version of Font Awesome.

### PR DESCRIPTION
**Pull request by nielslange**
Fixes #1278

**Steps to test:**

1. Head over to https://fontawesome.com/icons?d=gallery&v=5.12.1&m=free to look up all free icons that have been added with Font Awesome 5.12.1
2. Copy one of these icons, e.g. `<i class="fas fa-robot"></i>`
3. Create a new page and add the copied icon to the page title
4. Save the page and look up the result

**Result:**

<table>
<tr>
<td>

#1278-editor
</td>
<td>

#1278-frontend
</td>
</tr>
</table>

### CODE REVIEW

File: assets/css/base/icons.scss [hunk 1]
```diff
@@ -8,8 +8,11 @@
 // Sass tools and helpers used across the project.
 @import '../sass/utils/variables';
 @import '../sass/utils/mixins';
+@import '../sass/vendors/font-awesome/brands';
 @import '../sass/vendors/font-awesome/fontawesome';
+@import '../sass/vendors/font-awesome/regular';
 @import '../sass/vendors/font-awesome/solid';
+@import '../sass/vendors/font-awesome/v4-shims';
```
Comment (haszari) [inline assets/css/base/icons.scss:15]: Adding in all these other variations increases the size of this file (`icons.css`). When I built, the file grew from 99 KB (`master`) to 144KB on this branch - almost 45% increase.

I'd suggest we keep this focused on the variations we need for Storefront. If we take out `brands` `regular` and `v4-shims` then `icons.css` is 101KB.

Comment (nielslange) [reply assets/css/base/icons.scss:15]: Sounds good! I'll update this PR later today and will also update Font Awesome to version 5.13 as mentioned in https://github.com/woocommerce/storefront/pull/1279#pullrequestreview-396121519.

Comment (haszari) [reply assets/css/base/icons.scss]: I've taken care of these updates – thanks for your work on this @nielslange  🤝

**Comment by haszari** [comment]
Have retested this with latest changes and all is looking good - merging 🧜‍♀️

**Pull request merged**